src/libclipper/src/rpc_service.cpp

```diff
@@ -223,6 +235,22 @@ void RPCService::receive_message(
     vector<uint8_t> content((uint8_t *)msg_content.data(),
                             (uint8_t *)msg_content.data() + msg_content.size());
     RPCResponse response(id, content);
+
+    auto container_info_entry = containers.find(connection_id);
+    if(container_info_entry == containers.end()) {
+      throw std::runtime_error("Failed to find container that was previously registered via RPC");
+    }
+    std::pair<VersionedModelId, int> container_info = container_info_entry->second;
+
+    // TODO: These callbacks should be submitted to a threadpool for
+    // execution on separate threads
+    std::thread([&]() {
+      new_response_callback_(response);
+    }).detach();
+    std::thread([&]() {
+      container_ready_callback_(container_info.first, container_info.second);
+    }).detach();
```

Comment:
Instead of spawning a separate thread to execute these, just execute them directly in the thread of execution. It will be slow, but we will be adding a threadpool this week in #80.